src/HTMLElement/elements/HTMLTableElement.js

```diff
@@ -0,0 +1,309 @@
+import HTMLElement from '../../HTMLElement';
+import DOMException from '../../DOMException';
+
+/**
+ * The HTMLTable interface represents table
+ * and inherits all classes and methods of the HTMLElement interface.
+ *
+ * @see https://developer.mozilla.org/en/docs/Web/API/HTMLTableElement
+ * @class HTMLTableElement
+ */
+export default class HTMLTableElement extends HTMLElement {
+    constructor() {
+        super();
+    }
+
+    /**
+     * Caption of element, returns HTMLElement &lt;caption&gt;
+     * @member {HTMLElement} HTMLTableElement#caption
+     * @returns {HTMLElement}
+     */
+    get caption() {
+        return this.querySelector('caption');
+    }
+
+    /**
+     * Sets caption of element
+     * @param {HTMLElement} element - element with tagName caption
+     * @ignore
+     */
+    set caption(element) {
+        if (element.tagName !== 'caption') {
+            throw new DOMException('HierarchyRequestError');
+        }
+
+        let caption = this.caption;
+        if (caption !== null) {
+            this.replaceChild(element, caption);
+        } else {
+            if (this.firstChild) {
+                this.insertBefore(element, this.firstChild);
+            } else {
+                this.appendChild(element);
+            }
+        }
+    }
+
+    /**
+     * Table head
+     * @member {HTMLElement} HTMLTableElement#tHead
+     * @returns {HTMLElement}
+     */
+    get tHead() {
+        return this._childNodeFind(child => child.tagName === 'thead');
+    }
+
+    /**
+     * @ignore
+     * @param {HTMLElement} element
+     */
+    set tHead(element) {
+        if (element.tagName !== 'thead') {
+            throw new DOMException('HierarchyRequestError');
+        }
+
+        let thead = this.tHead;
+        if (thead !== null) {
+            this.replaceChild(element, thead);
+        } else {
+            if (this.children.every((child) => {
```

Comment:
> When set, if the object doesn't represent a `<thead>`, a DOMException with the HierarchyRequestError name is thrown. If a correct object is given, it is inserted in the tree immediately before the first element that is neither a <caption>, nor a `<colgroup>`, or as the last child if there is no such element, and the first `<thead>` that is a child of this element is removed from the tree, if any.

every is probably not the function you're looking for: `some` seems more appropriate

Do you have tests with cases like this ? Seems like something tricky